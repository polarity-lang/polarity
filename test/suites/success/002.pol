codata Fun(a: Type, b: Type) {
    Fun(a, b).ap(a: Type, b: Type, x: a) : b
}

codef Compose(a: Type, b: Type, c: Type, f: Fun(a, b), g: Fun(b, c)) : Fun(a, c) {
    ap(a', c', x) => g.ap(b, c, f.ap(a, b, x: a') : b) : c'
}
