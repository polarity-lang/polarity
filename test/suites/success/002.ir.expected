codef Compose(a, b, c, f, g) { .ap(a', c', x) => g.ap(b, c, f.ap(a, b, x)) }
